### DISCUSSION TITLE
Koneksi Jupyter Notebook ke ODBC 2

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#143] [Q&A] [answered]
Dear Tim Algoritma,

Mohon sarannya untuk membuat koneksi ODBC ke jupyter notebook, awalnya saya coba jalankan instalasi packages

# Virtual Environment
conda create --name caats python=3.9
conda activate caats
pip install pyodbc pymysql pandas ipykernel
python -m ipykernel install --user --name=caats

# Koneksi
import pyodbc
pyodbc.drivers()

server = 'xx.130.x.xxx'
database = 'library'
username = 'E90238xxx'
password = 'qwertyu4'

cnxn = pyodbc.connect('DRIVER={ODBC Driver 17 for SQL Server};SERVER='+server+';DATABASE='+database+';UID='+username+';PWD='+ password)

namun hasil yang saya dapatkan masih ada error
<img width="675" alt="image" src="https://user-images.githubusercontent.com/103427989/174516071-ea58b07a-6391-4c42-9ee7-63ce9eba6515.png">

mohon sarannya

Best regards,
JT

**Comment by dwiknrd** [comment, marked as answer]
Haloo [joejnt](https://github.com/joejnt)

Terima kasih sudah bertanya.

Apabila dilihat dari error yang dilampirkan disana terdapar error message "Login timeout expired". Hal tersebut kemungkinan terjadi karena server database yang bapak akses mengimplementasikan sistem white-list IP, dan IP bapak belum terdaftar sehingga tidak bisa melakukan koneksi ke server database tersebut. Untuk permasalah tersebut bapak dapat menghubungi database administrator yang bersangkutan.

Semoga bermanfaat :)
